frontend/src/pages/CreateVotingPage.jsx

```diff
@@ -0,0 +1,92 @@
+import css from "./CreateVotingPage.module.css";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+
+export function CreateVotingPage() {
+  const [fields, setFields] = useState([
+    { id: 1, value: "" },
+    { id: 2, value: "" },
+  ]);
+  const [convertedTime, setConvertedTime] = useState("");
+  const [clearCandidates, setClearCandidates] = useState([]);
+  const {
+    handleSubmit,
+    register,
+    formState: { isValid, errors },
+  } = useForm({
+    mode: "onBlur",
+  });
+
+  const onSubmit = function (data) {
+    const splitedTime = data.time.split(":").map(Number);
+    setConvertedTime(splitedTime[0] * 86400 + splitedTime[1] * 3600 + splitedTime[2] * 60 + splitedTime[3]);
+    setClearCandidates(data.candidates.filter(candidate => candidate));
+  };
+  const handleChange = function (id, value) {
+    const newFields = [...fields];
+    const index = newFields.findIndex(field => field.id === id);
+    newFields[index].value = value;
+    if (index === fields.length - 1 && index < 9 && value !== "") {
+      newFields.push({ id: fields.length + 1, value: "" });
```

Comment:
it seems like it's possible for a new field to have an id that was already used. You can avoid this by generating a new id randomly and checking if it exists, or using a ref counter like this 
```js
const [fields, setFields] = useState([
    { id: 1, value: "" },
    { id: 2, value: "" },
  ]);
const idCounterRef = useRef(3); //start from next available id
const handleChange = function (id, value) {
    newFields.push({ id: idCounterRef.current, value: "" });
    idCounterRef.current++;
 }
```